src/lib.rs

```diff
@@ -81,49 +82,43 @@ pub fn create_routes() -> impl Filter<Extract = impl warp::Reply, Error = warp::
     let generate_api = warp::path!("generate")
         .and(warp::post())
         .and(warp::body::bytes())
-        .map(move |body: warp::hyper::body::Bytes| {
-            // NOTE(ayubun): We parse JSON manually to handle malformed JSON as a 400 Bad Request.
-            // This decision was made because the default behaviour is to silently fallback to the
-            // empty body route, which generates 1 ID. I feel like this isn't as ergonomic as the
-            // API telling you that you've made an error loudly so that you can fix it.
-            let request: Option<GenerateRequest> = if body.is_empty() {
-                None
-            } else {
-                match serde_json::from_slice(&body) {
-                    Ok(req) => Some(req),
-                    Err(_) => {
-                        return warp::reply::with_status(
-                            "Invalid JSON format".to_string(),
-                            warp::http::StatusCode::BAD_REQUEST,
-                        );
+        .and_then(move |body: warp::hyper::body::Bytes| {
+            let snowflake_generator = snowflake_generator.clone();
+            async move {
```

Comment:
hmm .. what is the benefit for all of this nesting? does this async move actually provide potential performance gains? it would be worth benchmarking, because i think the code is a lot less clean with it

my assumption was that warp manages the tokio runtime for us, so we do not need to worry about spawning our own async threads. if that is true, this async move is out of place